### ISSUE TITLE
Logarithmic expression on a decision variable

### ISSUE THREAD

**Issue by diegorossit** [scipopt/JSCIPOpt#30]
Hi!

I would like to implement a constraint that has a logarithmic expression applied on a decision variable. I think we should be able to this follwoing instructions in [https://github.com/scipopt/JSCIPOpt#readme](url) but this is quite complex for beginners. Any suggestion? I see that in the Python [PySCIPOpt](https://github.com/scipopt/PySCIPOpt) is already implemented.

Thanks in advance for your time.

**Comment by ambros-gleixner** [comment]
Agreed, but this would be the only way to do so via JSCIPOpt. If you are motivated to build this extension, it is a good idea to look at the PySCIPOpt implementation.

**Comment by diegorossit** [comment]
Thanks, I will give a try.

**Comment by diegorossit** [comment]
Hi, in this regard do you know if regarded I implement the logarithmic expression I can optimize an mixed-integer problem with an objective function of the form: x1 * log (1 + x2) with SCIP? (in which x1 and x2 are two non-negative continuous variable). 

I mean building a logarithmic expression and then multiplying that with a variable in a .... quadratic expression (?).

Thanks in advance for your time.

**Comment by diegorossit** [comment]
Hi!

Considering I implement the logarithmic expression, do you know if I would be able to optimize a mixed-integer problem with an objective function of the form x1 * log(1 + x2)? in which x1 and x2 are two continuous positive variables. This would imply building the logarithmic expression "log(1 + x2)" and then adding it in the "createConsQuadratic" function as an element to multiply the term "x1".

Thanks in advance for your kind time.

**Comment by diegorossit** [comment]
Thanks for your help. I looked at the files and PySCIPOpt has a type SCIP_EXPR which is not implemented in JSCIPOpt. This type is used in both SCIPcreateExprLog and SCIPcreateConsNonlinear. So it's harder than I thought.  
I think I'll wait that more skilled collaborators can extend it.

Anyway, thanks for your kind help and time. Regards!
